File: JavaDemoGithubPractice/app/src/main/java/com/example/javademogithubpractice/inject/component/AppComponent.java


package com.example.javademogithubpractice.inject.component;



import com.example.javademogithubpractice.AppApplication;
import com.example.javademogithubpractice.dao.DaoSession;
import com.example.javademogithubpractice.inject.module.AppModule;

import javax.inject.Singleton;

import dagger.Component;


@Singleton
@Component(modules = {AppModule.class})
public interface AppComponent {
    AppApplication getApplication();
    DaoSession getDaoSession();

}

File: KotlinDemoGithubProject/app/src/main/java/com/example/kotlindemogithubproject/inject/component/AppComponent.kt
package com.example.kotlindemogithubproject.inject.component


import com.example.kotlindemogithubproject.AppApplication
import com.example.kotlindemogithubproject.dao.DaoSession
import com.example.kotlindemogithubproject.inject.module.AppModule
import dagger.Component

import javax.inject.Singleton


@Singleton
@Component(modules = [AppModule::class])
interface AppComponent {
    val application: AppApplication
    val daoSession: DaoSession

}
